form-regex.rst

```diff
@@ -0,0 +1,89 @@
+Using regular expressions
+============================
+
```

Comment:
I think some context at the beginning about where regex is used in odk would be helpful.